core/src/main/java/fr/sncf/osrd/api/InfraManager.java

```diff
@@ -240,8 +240,7 @@ public InfraCacheEntry deleteFromInfraCache(String infraId) {
     public FullInfra getInfra(String infraId, String expectedVersion, DiagnosticRecorder diagnosticRecorder)
             throws OSRDError, InterruptedException {
         try {
-            infraCache.putIfAbsent(infraId, new InfraCacheEntry());
-            var cacheEntry = infraCache.get(infraId);
+            var cacheEntry = infraCache.getOrDefault(infraId, new InfraCacheEntry());
```

Comment:
I don't think that's right. Sure, it's better than what was there before, but not great either: if the new InfraCacheEntry is used:

 - either loadIfMissing is true, and it's just discarded
 - or elaborate checks run on this cache entry, which is just empty

I would rather do something like
```java
            var cacheEntry = infraCache.get(infraId);
            if (cacheEntry == null)
                // return an error or something
```